onc-healthit/lantern-back-end
# Query well known endpoints

**Changes made this PR**

-  Capability querier queries the /.well-known/smart-configuration endpoint in addition to the /metadata 

- http status and response from well known endpoint added to Fhir_endpoints_info db 

- Small changes to message struct and FhirEndpointInfo struct 

**To test**

- `make run`
- populate db with endpoints `make populatedb`
- Send the endpoints to the queues: in `endpointmanager/cmd`, run `go run sendendpoints/main.go`
- Run the capability receiver: in `endpointmanager/cmd`, run `go run capabilityreceiver/main.go`
- Verify that the fhir_endpoints_info table in db has columns smart_http_response and smart_response
- Verify that some of the smart_http_response have 200 and that some of the smart_response have actual statements

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: 
https://oncprojectracking.healthit.gov/support/browse/LANTERN-199
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
